src/Neo/SmartContract/Native/Notary.cs

```diff
@@ -0,0 +1,336 @@
+        /// <returns>Deposit lock height of the specified address.</returns>
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
+        public uint ExpirationOf(DataCache snapshot, UInt160 acc)
+        {
+            Deposit deposit = GetDepositFor(snapshot, acc);
+            if (deposit is null) return 0;
+            return deposit.Till;
+        }
+
+        /// <summary>
+        /// BalanceOf returns deposited GAS amount for specified address.
+        /// </summary>
+        /// <param name="snapshot">DataCache</param>
+        /// <param name="acc">Account</param>
+        /// <returns>Deposit balance of the specified account.</returns>
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
+        public BigInteger BalanceOf(DataCache snapshot, UInt160 acc)
+        {
+            Deposit deposit = GetDepositFor(snapshot, acc);
+            if (deposit is null) return 0;
+            return deposit.Amount;
+        }
+
+        /// <summary>
+        /// Withdraw sends all deposited GAS for "from" address to "to" address. If "to"
+        /// address is not specified, then "from" will be used as a sender.
+        /// </summary>
+        /// <param name="engine">ApplicationEngine</param>
+        /// <param name="from">From Account</param>
+        /// <param name="to">To Account</param>
+        /// <returns>Whether withdrawal was successfull.</returns>
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.States)]
+        private async ContractTask<bool> Withdraw(ApplicationEngine engine, UInt160 from, UInt160 to)
+        {
+            if (!engine.CheckWitnessInternal(from)) throw new InvalidOperationException(string.Format("Failed to check witness for {0}", from.ToString()));
+            var receive = to is null ? from : to;
+            Deposit deposit = GetDepositFor(engine.Snapshot, from);
+            if (deposit is null) throw new InvalidOperationException(string.Format("Deposit of {0} is null", from.ToString()));
+            if (Ledger.CurrentIndex(engine.Snapshot) < deposit.Till) throw new InvalidOperationException(string.Format("Can't withdraw before {0}", deposit.Till));
+            RemoveDepositFor(engine.Snapshot, from);
+
+            await engine.CallFromNativeContract(Hash, GAS.Hash, "transfer", Hash.ToArray(), receive.ToArray(), deposit.Amount, null);
+            return true;
+        }
+
+        /// <summary>
+        /// GetMaxNotValidBeforeDelta is Notary contract method and returns the maximum NotValidBefore delta.
+        /// </summary>
+        /// <param name="snapshot">DataCache</param>
+        /// <returns>NotValidBefore</returns>
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
+        public uint GetMaxNotValidBeforeDelta(DataCache snapshot)
+        {
+            return (uint)(BigInteger)snapshot[CreateStorageKey(Prefix_MaxNotValidBeforeDelta)];
```

Comment:
```csharp
var bi = (BigInteger)snapshot[CreateStorageKey(Prefix_MaxNotValidBeforeDelta)];
if (uint.TryParse(bi.ToString(), out var result))
    return result;
else
    // TODO: Add line for error handling for overflow.
```

Reply:
Not needed, maximum bound is checked in setter.